appengine/flexible/static_files_spa/main.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# [START gae_flex_python_static_files_spa]
+import os
+import logging
+
+from flask import Flask, send_from_directory
+
+# folder for static content relative to the current module
+STATIC_DIR = os.getenv('STATIC_DIR') or 'static'
+
+app = Flask(__name__)
+
+
+@app.route('/', defaults={'path': ''})
+@app.route('/<path:path>')
+def catch_all(path):
+    # NOTE: we don't use Flusk standard support for static files
+    # (static_folder option and send_static_file method)
+    # because they can't distinguish requests for static files (js/css) 
+    # and client routes (like /products)
```

Comment:
I don't understand this comment. This code is serving static files, isn't it? Why not let Flask, or better yet, App Engine, do that for the app instead?

Reply:
During client-side routing in Single-Page Apps, such as built with Angular or any other FWs, you'll have page urls on the client that do not correlate to any server-side routes. If the user reloads a page with such an url your normal static-files Flask handler or built-in GAE won't work. Because they won't understand what to do with such urls.

Can you please run the sample. Then you can replicate client-side navigation in the [existing](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/appengine/flexible/static_files) "static-files" sample and see that on page reload you're getting "NotFound".